spark/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala

```diff
@@ -2398,6 +2398,18 @@ trait DeltaErrorsBase
     new DeltaAnalysisException(errorClass = "DELTA_UNSET_NON_EXISTENT_PROPERTY", Array(key, table))
   }
 
+  def identityColumnWithGenerationExpression(): Throwable = {
+    new AnalysisException("Identity column cannot be specified with a generated column expression.")
+  }
+
+  def identityColumnIllegalStep(): Throwable = {
+    new AnalysisException("Identity column step cannot be 0.")
+  }
+
+  def identityColumnDataTypeNotSupported(): Throwable = {
+    new AnalysisException("Identity column does not support this data type.")
```

Comment:
It would be nice to tell the user what is the supported type and the unsupported type :)

Reply:
I checked and it seems the other existing error message don't provide the list of supported types. It would be more consistent to not report the supported types.

Reply:
That's OK.